cpp-package/src/inc/brainflow_array.h

```diff
@@ -0,0 +1,468 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <iostream>
+#include <memory>
+#include <stdlib.h>
+#include <string.h>
+
+#include "brainflow_constants.h"
+#include "brainflow_exception.h"
+
+
+template <typename T, size_t Dim>
+class BrainFlowArray;
+template <typename T, size_t Dim>
+std::ostream &operator<< (std::ostream &os, const BrainFlowArray<T, Dim> &array);
+
+template <size_t N>
+static inline int product (const std::array<int, N> &array)
+{
+    int result = 1;
+
+    for (int i = 0; i < N; i++)
+    {
+        result *= array[i];
+    }
+
+    return result;
+}
+
+template <size_t N>
+static inline std::array<int, N> zero_array ()
+{
+    std::array<int, N> result;
+
+    for (int i = 0; i < N; i++)
```

Comment:
If you wanted, you could use [`result.fill(0)`](https://en.cppreference.com/w/cpp/container/array/fill) here instead of a for loop.  This would let the standard library use an implementation optimised for the platform.